doc/lib/mqtt.txt

```diff
@@ -0,0 +1,60 @@
+/**
+@mainpage
+@anchor mqtt
+@brief MQTT 3.1.1 client library.
+
+MQTT is the protocol that allows a device to communicate with AWS IoT. The MQTT library provided with this SDK implements a subset of the MQTT v3.1.1 protocol. The sample application for Linux `subscribe_publish_sample.c` is provided as a starting point for using this library.
+
+The interface for communication over MQTT is provided in the file `aws_iot_mqtt_interface.h`.
+- MQTT session creation: @ref mqtt_function_init and @ref mqtt_function_free
```

Comment:
Should this be "MQTT library management"? How is the init and free associated with MQTT session creation?

Reply:
These functions are specifically for creating and destroying an MQTT client context.  I've reworded it because that's the term used elsewhere.